docs/howto/craft-for-platforms/example-ros-2-app.rst

```diff
@@ -0,0 +1,341 @@
+packages across multiple snaps, saving space and ensuring package consistency
+throughout your snap build environment.
+
+You can find more information in `ROS architectures with snaps
+<https://ubuntu.com/robotics/docs/ros-architectures-with-snaps>`_ in the
+Canonical ROS documentation.
+
+Turning on content sharing takes requires two small changes in the recipe.
+Here's the difference in the recipe when content sharing is enabled:
+
+.. tabs::
+
+  .. group-tab:: core20
+
+    .. code:: diff
+
+      source-subdir: demo_nodes_cpp
+      -  stage-packages: [ros-foxy-ros2launch]
+
+      apps:
+        ros2-talker-listener:
+          command: ros2 launch demo_nodes_cpp talker_listener.launch.py
+      -   extensions: [ros2-foxy]
+      +   extensions: [ros2-foxy-ros-base]
+
+  .. group-tab:: core22
+
+    .. code:: diff
+
+      source-subdir: demo_nodes_cpp
+      -  stage-packages: [ros-humble-ros2launch]
+
+      apps:
+        ros2-talker-listener:
+          command: ros2 launch demo_nodes_cpp talker_listener.launch.py
+      -   extensions: [ros2-humble]
+      +   extensions: [ros2-humble-ros-base]
+
+  .. group-tab:: core24
+
+    .. code:: diff
+
+      source-subdir: demo_nodes_cpp
+      -  stage-packages: [ros-jazzy-ros2launch]
+
+      apps:
+        ros2-talker-listener:
+          command: ros2 launch demo_nodes_cpp talker_listener.launch.py
+      -   extensions: [ros2-jazzy]
+      +   extensions: [ros2-jazzy-ros-base]
+
+To turn on content sharing:
+
+#. Remove the ``stage-packages`` key from the part. The package is already available in the
+   content-sharing snap.
+#. Change the ROS 2 extensions in ``extensions`` to the variant that correspond
+   to the core:
+
+   .. list-table::
+      :header-rows: 1
+
+      * - Core
+        - Content extension
+      * - core20
+        - `ros-foxy-ros-core
+          <https://snapcraft.io/docs/ros2-foxy-content-extension>`_
+      * - core22
+        - `ros2-humble-ros-base
+          <https://snapcraft.io/docs/ros2-humble-content-extension>`_
+      * - core24
+        - `ros2-jazzy-ros-base
+          <https://snapcraft.io/docs/ros2-jazzy-content-extension>`_
+
+
+Before launching the app for the first time, you must connect the .
+Since our snap makes use of the content provided by another snap, we have to
+connect them.
+
+To connect the snaps:
+
+#. Run:
+
+   .. code:: bash
+
+     snap connect ros2-talker-listener:ros-foxy ros-foxy-ros-base
+
+#. Verify that the connection is established by running:
+
+   .. code:: bash
+
+     snap connections ros2-talker-listener
+
+   If the connection is successful, the output will show that through the
+   content interface, the snap's ROS launch command is manually plugged in to
+   the ROS base snap.
+
+Lastly, the content-sharing extension is experimental. Before you can build a
+snap that uses it, you must enable experimental extensions in your terminal:
+
+.. code:: bash
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=1
```

Comment:
This can be dropped as it's becoming stable this month: https://github.com/canonical/snapcraft/pull/5156

Reply:
1684dd99af751120f065d1c021d354e9ab201bf4